src/main/java/com/direwolf20/mininggadgets/common/items/gadget/MiningCollect.java

```diff
@@ -40,15 +40,36 @@ public static List<BlockPos> collect(Player player, BlockHitResult startBlock, L
         Direction right = vertical ? up.getClockWise() : side.getCounterClockWise();
         Direction left = right.getOpposite();
 
-        coordinates.add(startPos.relative(up).relative(left));
-        coordinates.add(startPos.relative(up));
-        coordinates.add(startPos.relative(up).relative(right));
-        coordinates.add(startPos.relative(left));
-        coordinates.add(startPos);
-        coordinates.add(startPos.relative(right));
-        coordinates.add(startPos.relative(down).relative(left));
-        coordinates.add(startPos.relative(down));
-        coordinates.add(startPos.relative(down).relative(right));
+        if (range >= 3) {
+            coordinates.add(startPos.relative(up).relative(left));
+            coordinates.add(startPos.relative(up));
+            coordinates.add(startPos.relative(up).relative(right));
+            coordinates.add(startPos.relative(left));
+            coordinates.add(startPos);
+            coordinates.add(startPos.relative(right));
+            coordinates.add(startPos.relative(down).relative(left));
+            coordinates.add(startPos.relative(down));
+            coordinates.add(startPos.relative(down).relative(right));
+        }
+
+        if (range >= 5) {
```

Comment:
Should note here is this will do a 5 x 5 from the middle of the laser.
`XXXXX
XXXXX
XXOXX
XXXXX
XXXXX`

It will not make an easy tunnel for mining, if this was to do a nice mining tunnel you would have to do a 5x5 with 3 up, 2 down, and 2 to the sides.

`XXXXX
XXXXX
XXXXX
XXOXX
XXXXX`

Reply:
Play tested, 3 up and 1 down is sooo much better for tunnel mining, so this is what's in place now.

Maybe mak this configurable via the settings UI if you have 5x5 set?